src/main/java/uk/gov/hmcts/darts/common/entity/CourthouseEntity.java

```diff
@@ -49,7 +50,33 @@ public class CourthouseEntity extends CreatedModifiedBaseEntity {
         inverseJoinColumns = {@JoinColumn(name = "grp_id")})
     private Set<SecurityGroupEntity> securityGroups = new LinkedHashSet<>();
 
+    @ManyToMany
+    @JoinTable(name = "courthouse_region_ae",
+        joinColumns = {@JoinColumn(name = "cth_id")},
+        inverseJoinColumns = {@JoinColumn(name = "reg_id")})
+    private Set<RegionEntity> regions = new LinkedHashSet<>();
+
     @Column(name = "display_name")
     private String displayName;
 
+    public RegionEntity getRegion() throws IllegalStateException {
+
+        if (this.regions != null && this.regions.size() > 1) {
+            throw new IllegalStateException();
+        } else if (this.regions == null || this.regions.isEmpty()) {
```

Comment:
apache commons methods are normally null safe, so this can be replaced with the following to improve readability:-
`else if (CollectionUtils.isNotEmpty(regions))`

Reply:
updated